services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/modules/db/repositories/resource_tracker.py

```diff
@@ -183,118 +169,89 @@ async def update_service_run_stopped_at(
-        row = result.first()
-        return cast(PositiveInt, row[0]) if row else 0
-
-    async def list_service_runs_by_user_and_product_and_wallet(
+    async def list_service_runs_by_product_and_user_and_wallet(
         self,
-        user_id: UserID,
         product_name: ProductName,
-        wallet_id: WalletID,
+        user_id: UserID | None,
+        wallet_id: WalletID | None,
         offset: int,
         limit: int,
-    ) -> list[ServiceRunDB]:
+    ) -> list[ServiceRunWithCreditsDB]:
         async with self.db_engine.begin() as conn:
             query = (
-                self._service_runs_select_stmt()
-                .where(
-                    (resource_tracker_service_runs.c.user_id == user_id)
-                    & (resource_tracker_service_runs.c.product_name == product_name)
-                    & (resource_tracker_service_runs.c.wallet_id == wallet_id)
+                sa.select(
+                    resource_tracker_service_runs.c.product_name,
+                    resource_tracker_service_runs.c.service_run_id,
+                    resource_tracker_service_runs.c.wallet_id,
+                    resource_tracker_service_runs.c.wallet_name,
+                    resource_tracker_service_runs.c.pricing_plan_id,
+                    resource_tracker_service_runs.c.pricing_unit_id,
+                    resource_tracker_service_runs.c.pricing_unit_cost_id,
+                    resource_tracker_service_runs.c.pricing_unit_cost,
+                    resource_tracker_service_runs.c.user_id,
+                    resource_tracker_service_runs.c.user_email,
+                    resource_tracker_service_runs.c.project_id,
+                    resource_tracker_service_runs.c.project_name,
+                    resource_tracker_service_runs.c.node_id,
+                    resource_tracker_service_runs.c.node_name,
+                    resource_tracker_service_runs.c.service_key,
+                    resource_tracker_service_runs.c.service_version,
+                    resource_tracker_service_runs.c.service_type,
+                    resource_tracker_service_runs.c.service_resources,
+                    resource_tracker_service_runs.c.started_at,
+                    resource_tracker_service_runs.c.stopped_at,
+                    resource_tracker_service_runs.c.service_run_status,
+                    resource_tracker_service_runs.c.modified,
+                    resource_tracker_service_runs.c.last_heartbeat_at,
+                    resource_tracker_credit_transactions.c.osparc_credits,
+                    resource_tracker_credit_transactions.c.transaction_status,
                 )
+                .select_from(
+                    resource_tracker_service_runs.join(
+                        resource_tracker_credit_transactions,
+                        resource_tracker_service_runs.c.service_run_id
+                        == resource_tracker_credit_transactions.c.service_run_id,
```

Comment:
Just checking, but these are both indexed in both tables?

Reply:
correct both are indexed :+1: